src/include/http.h

```diff
@@ -46,112 +48,162 @@ extern "C" {
+ * Defines shared HTTP message content
+ */
+struct http_payload
+{
+   struct deque* headers;  /**< HTTP headers as deque */
+   void* data;             /**< Message data */
+   size_t data_size;       /**< Size of message data */
+};
+
+/** @struct http_request
+ * Defines a HTTP request
+ */
+struct http_request
+{
+   struct http_payload payload; /**< Request payload */
+   int method;                  /**< HTTP method */
+   char* path;                  /**< Request path */
+};
+
+/** @struct http_response
+ * Defines a HTTP response
+ */
+struct http_response
+{
+   struct http_payload payload; /**< Response payload */
+   int status_code;             /**< HTTP status code */
+};
+
 /** @struct http
- * Defines a HTTP interaction
+ * Defines a HTTP connection
  */
 struct http
 {
-   int endpoint;           /**< The endpoint */
-   int socket;             /**< The socket descriptor */
-   char* body;             /**< The HTTP response body */
-   char* headers;          /**< The HTTP response headers */
-   char* request_headers;  /**< The HTTP request headers */
-   SSL* ssl;               /**< The SSL connection (NULL for non-secure) */
+   int socket;      /**< The socket descriptor */
+   SSL* ssl;        /**< The SSL connection (NULL for non-secure) */
+   char* hostname;  /**< The hostname */
+   int port;        /**< The port number */
+   bool secure;     /**< Use SSL if true */
 };
 
 /**
- * Connect to an HTTP/HTTPS server
+ * Create a connection to a HTTP/HTTPS server
  * @param hostname The host to connect to
  * @param port The port number
  * @param secure Use SSL if true
- * @param result The resulting HTTP structure
- * @return 0 upon success, otherwise 1
+ * @param result The resulting HTTP connection
+ * @return PGMONETA_HTTP_STATUS_OK upon success, otherwise PGMONETA_HTTP_STATUS_ERROR
  */
 int
-pgmoneta_http_connect(char* hostname, int port, bool secure, struct http** result);
+pgmoneta_http_create(char* hostname, int port, bool secure, struct http** result);
 
 /**
- * Disconnect and clean up HTTP resources
- * @param http The HTTP structure
- * @return 0 upon success, otherwise 1
+ * Create a HTTP request
+ * @param method The HTTP method
+ * @param path The request path
+ * @param result The resulting HTTP request
+ * @return PGMONETA_HTTP_STATUS_OK upon success, otherwise PGMONETA_HTTP_STATUS_ERROR
  */
 int
-pgmoneta_http_disconnect(struct http* http);
+pgmoneta_http_request_create(int method, char* path, struct http_request** result);
 
 /**
  * Add a header to the HTTP request
- * @param http The HTTP structure
+ * @param request The HTTP request
  * @param name The header name
  * @param value The header value
- * @return 0 upon success, otherwise 1
+ * @return PGMONETA_HTTP_STATUS_OK upon success, otherwise PGMONETA_HTTP_STATUS_ERROR
  */
 int
-pgmoneta_http_add_header(struct http* http, char* name, char* value);
+pgmoneta_http_add_header(struct http_request* request, char* name, char* value);
```

Comment:
We could these `pgmoneta_http_request_add_header` and so on, and add a `pgmoneta_http_response_get_header`